Enable a time restriction to avoid new accounts to create answers

denphi opened the pull request:
![Screenshot 2025-03-20 at 11 29 50 AM](https://github.com/user-attachments/assets/17b22cf1-f957-4007-9561-585c30e190a1)

nkissebe committed:
[behaviors] add html behavior support for alpinejs, bootstrap, htmx

nkissebe committed:
[loader] change layout not found to a 404 error instead of 500

nkissebe committed:
[loader] allow task methods to be __call()'d

nkissebe committed:
[loader] create a default minimal function site controller class with dynamic Task support

nkissebe committed:
[loader] create a default minimal function site router class

nkissebe committed:
[loader] if no component entrypoint use a default

nkissebe committed:
[lucent] add copy of lucent template to core

nkissebe committed:
[lucent] fix old style EXEC check to use _HZEXEC_

nkissebe committed:
[lucent] rename classname ariaControlID to aria-control-target- for aesthetics

nkissebe committed:
[lucent] remove mod_menu override in lucent, functionality moved to core

nkissebe committed:
[lucent] remove readme from lucent template since we are now in core

nkissebe committed:
[lucent] add raw layout option to mod_custom module

nkissebe committed:
[lucent] remove mod_custom view override from lucent, functionality now in core (raw layout)

nkissebe committed:
[lucent] remove com_search view override from lucent, functionality now in core

nkissebe committed:
[lucent] promote css overflow property from lucent override to core

nkissebe committed:
[lucent] remove com_projects view override from lucent, functionality now in core

nkissebe committed:
[lucent] promote removal of icon from introduction css from lucent override to core

nkissebe committed:
[lucent] remove system (introduction) css file from lucent, functionality now in core

nkissebe committed:
[lucent] promote com_search overrides from lucent to core

nkissebe committed:
[ratelimit] start ratelimit exploration. currently just a hit count (WIP)

nkissebe committed:
[ratelimit] add hit and time counters to APCu for future use (WIP)

nkissebe committed:
[lucent] move mod_menu override disclosure menus from lucent to core

nkissebe committed:
[lucent] conditionally wrap the inner div for disclosure menus, not sure this is best

nkissebe committed:
[php >=8.2][splitbrain/lesserphp] switch lesserphp repos again for php 8.x fixes

nkissebe committed:
[php >=8.2] fix lessc usage to use new splitbrain fork of lesserphp

nkissebe committed:
[php >=8.2] fix pucas logging and constructor for latest cas library version

nkissebe committed:
[php >=8.2][guzzlehttp/guzzle] update guzzle package to 7.9.*

nkissebe committed:
don't do Purdue CAS logout if not logged in via Purdue CAS

nkissebe committed:
correct references to sym links, should be just [hard] links

nkissebe committed:
avoid deprecation and error supression notices in log

cilogon library uses error supression to avoid making
manual checks for array key existence which is slower
and pollutes our agressive error logging. we now bypass
their helper functions and check response array directly.

nkissebe committed:
[league/oauth2-client] update oauth2-client to 2.8.1 to fix now reverted scope option bug

nkissebe commented:
I'll review this next week and merge. But as an end user I hate this. I go to a site to find something out and want to ask a question/post to forum so I create an account then find out I can't until X time passes or something. I am looking for an answer right away and this just turns me off the site for good.

denphi commented:
You can enable or disable it for each hub. I also dislike it, but this answers feature serves as an easy spam endpoint (e.g., https://stemedhub.org/answers).

nkissebe commented:
> You can enable or disable it for each hub

That's why I'll merge it 😉

nkissebe committed:
fix down() method of migration

nkissebe committed:
disambiguate getVar() name

nkissebe committed:
disambiguate getVar/setVar methods in Uri class

nkissebe committed:
filter out invalid (array) url route elements

nkissebe committed:
use Request:: facade instead of accessing via app

nkissebe committed:
always fetch option with getCmd instead of getVar

nkissebe committed:
add Request::getSimpleArray() to validate simple array of scalars

nkissebe committed:
use getSimpleArray() instead of getArray() to sanitize input

nkissebe committed:
[HZ-1080] escape search string

dbenham committed:
Update edit.php

php8 related fix in the user import scripts

dbenham committed:
Update Reader.php

php8 update related to user import

nkissebe committed:
catch correct exception class

nkissebe committed:
don't pass null to trim here

nkissebe committed:
fail user save if no username set

nkissebe committed:
don't trigger sendSolrRequest on empty models

dbenham committed:
Update formathtml.php

php8 update

dbenham committed:
Update externalhref.php

php8 update

dbenham committed:
Update renderer.php

php8 update

denphi committed:
Small typo fix

Colon is already added in the resources view, should not have to be included in the language definition

denphi committed:
Overwriting styles

multi-pages quizzes overwrite styles

denphi committed:
details.php

Overwriting styles when correct/incorrect questions displayed

Commit:
fix for XSS during new user registration

Commit:
fix for XSS for user name change

Commit:
fix for nH ticket 435022 new tool creation

jsperhac committed:
[HZ-1046] Add columns and data to exclude_list metrics table.

denphi committed:
Title incorrect if ticket does not exist

Title incorrect if ticket does not exist

denphi committed:
Update entry.php

Preview page is not ordered as it supposed to be

nkissebe committed:
disable writing to ratelimit table until purge logic is committed

nkissebe committed:
default some parameters to '' instead of null

nkissebe committed:
convert string to int before doing math

nkissebe committed:
remove deprecated call to set auto_detect_line_endings

nkissebe committed:
check if array before counting

nkissebe committed:
fix null check properly this time

jsperhac committed:
[NAIRR-162] Improve category display text for search results

jsperhac committed:
[NAIRR-50] Only single slashes in url for uncategorized Content items

denphi committed:
non integer error

force int values passed to abs function

denphi committed:
deprecated calls

JUtility was deprecated

denphi committed:
another call

denphi committed:
Validate that NULL is not passed

Rows2 can be defined as null and can be passed to the template if there are no objects

denphi committed:
incorrect use of "setter"

default values should be set by using "getters" functions not

nkissebe committed:
fix function name

nkissebe committed:
another set off null to '' fixes

nkissebe committed:
fix variable name

nkissebe committed:
add missing class properties

nkissebe committed:
fix more null to '' issues

nkissebe committed:
fix null to '' errors in com_installer

nkissebe committed:
check existence before use

nkissebe committed:
use the User facade here

nkissebe committed:
null to '' check in profile plugin

nkissebe committed:
fix password field name/id

jsperhac committed:
Check content publish-up and publish-down dates when Solr indexing

nkissebe committed:
include check for variable existance

nkissebe committed:
some null to '' fixes from purr

nkissebe committed:
is_array/count fixes from purr

nkissebe committed:
fix setPriority() method to use new base mail class

nkissebe committed:
add prm file extension to mimetypes list

jsperhac committed:
[NAIRR-164] Use correct Solarium object in method signature.

nkissebe committed:
geodynamics: order resources by asc explicitly

nkissebe committed:
php 8 fixes from cdmhub

nkissebe committed:
fix more symLink call references

nkissebe committed:
dev.purr null to '' fixes

mdlama committed:
Undefined variable in view

mdlama committed:
Create initialized prop to avoid null

Avoids passing null to explode
Hopefully this general fix will avoid this issue for all macros
  that inherit the class.

mdlama committed:
Missing closed parenthesis

mdlama committed:
Partial revert uploaded extensions fix

mdlama committed:
Avoid passing null to preg_split, preg_match_all

mdlama committed:
is_null to empty

Wiki special pages weren't showing

mdlama committed:
Avoid passing null to preg_replace

mdlama committed:
Avoid passing null to stripslashes

mdlama committed:
Null to '' fix

mdlama committed:
Avoid passing null to preg_match_all

mdlama committed:
Avoid passing null to trim

jsperhac committed:
Remove outdated Solarium buffer hack

mdlama committed:
Avoid dynamic property issue

(cherry picked from commit 0ec192206aba7f401658cfc79e7d116e60e58de2)

mdlama committed:
Avoid null to strtotime

(cherry picked from commit a60fde6306a4f5ba9a321959df4473f076fd174e)

mdlama committed:
Fixed newsletter controller route logic

Also removed token requirement from subscribe task.

(cherry picked from commit a6cac3ecac526d271c4834a6a3ea663aa0082ce3)

nkissebe committed:
null to '' fixes in com_support

nkissebe committed:
more null to '' fixes in com_events

nkissebe committed:
fix inversion of conditional when deciding to set/add to: address

nkissebe committed:
fix variable syntax error in likes

nkissebe committed:
restore checkToken for POSTs

nkissebe committed:
[PR #1795] Adding error handling when muse returns NULL

mdlama committed:
Changed protected class to public

(cherry picked from commit 74d1a49f1ebad2e8f931c8afc4c508898d281532)

mdlama committed:
Avoid passing null to trim

(cherry picked from commit 98cb298fb991b734997f6e5462401058bcf99852)

nkissebe committed:
countable check

mdlama committed:
Init macro args to empty string to avoid null

nkissebe committed:
convert strings to integers before doing math

nkissebe committed:
use \w instead of . to avoid pcre back limit

nkissebe committed:
null to '' issues found on qubes

nkissebe committed:
check if directory exists before using

nkissebe committed:
check aclgroup result before using

nkissebe committed:
convert strings to integer before doing math

nkissebe committed:
null to '' issue on theghub

nkissebe committed:
another null '' for qubes

mdlama committed:
Check not dir in exif_imagetype

(cherry picked from commit ec198427fa6ae4d482b086a4b710dfe5e6c9edc6)

mdlama committed:
Avoid sending null to explode

(cherry picked from commit c50fa7f385b43ed4511c620b20784db90ff66a4f)

mdlama committed:
Macro redirect args avoids @list

(cherry picked from commit e2f95da5a63c3ac766a9704f6e5af862700ae139)

mdlama committed:
Make sure detail exists for new ticket

(cherry picked from commit 1d678fb10a3fd4368d61452543ab1a551fe9b011)

nkissebe committed:
make error message more explicit

nkissebe committed:
encode special characters in url

nkissebe committed:
null to '' fix from nanohub

nkissebe committed:
convert string to float before doing math

nkissebe committed:
more null to '' issues from qubeshub

nkissebe committed:
array check for count()

nkissebe committed:
update router to use routing class

nkissebe committed:
scan for newer wkhtmltopdf calling format/location

nkissebe committed:
ensure name is a string type in setFrom()

nkissebe committed:
com_kb null to '' fix

nkissebe committed:
fix comment

nkissebe committed:
remove comment

nkissebe committed:
update component router to use newer router class

nkissebe committed:
com_cart null to '' fix

nkissebe committed:
add missing class property

nkissebe committed:
check group result before using

nkissebe committed:
use from rather than sender

nkissebe committed:
make sure accessGroupScope is an array

nkissebe committed:
explicitly look for request as array

nkissebe committed:
add missing class property

nkissebe committed:
com_dataviewer null to '' issue

nkissebe committed:
check if directory before using

nkissebe committed:
update reference to JUtility

mdlama committed:
Change require_once from relative to absolute path

Causes an issue with app overrides (relative path loads from core, but
later in com_members/models/registration.php loads from app,
causing conflict).

mdlama committed:
leftJoin requires 3 arguments

nkissebe committed:
mod_whatsnews null to '' issue

nkissebe committed:
array check before foreach

nkissebe committed:
in php 8 0 != '' so using 0 here would produce an empty list later

nkissebe committed:
add missing class property

nkissebe committed:
fix orcid temporary user login

nkissebe committed:
explicitly convert to integer before multiplying

nkissebe committed:
com_installer null to '' fixes

nkissebe committed:
add missing class property

nkissebe committed:
some nanohub php 8 fixes

denphi committed:
Update instructorprogress.js

recognitions has integers

denphi committed:
Update resources.php

added array validation

nkissebe committed:
null to '' issues in com_support

nkissebe committed:
don't pass database to maxwell client yet

nkissebe committed:
more don't pass database to maxwell client yet, and null to '' fix

nkissebe committed:
default value to '' instead of null

nkissebe committed:
null to '' issue in googlescholar plugin

nkissebe committed:
remove favorites from the favorites tabs without searching all the tools tabs

nkissebe committed:
default values to '' instead of null

denphi committed:
Update watch.php

File detection should be relative to app, not to root

nkissebe committed:
don't rely on false to array conversion

nkissebe committed:
null to '' issue in dublincore plugin

nkissebe committed:
hotfix to default css

nkissebe committed:
fix sql statement, quote and spacing

nkissebe committed:
hotfix to kameleon template access colors

nkissebe committed:
add some missing class properties

nkissebe committed:
default more values to '' instead of null

nkissebe committed:
handle error a little better

nkissebe committed:
ensure interests is an array

nkissebe committed:
set variable before trying to append to it

nkissebe committed:
in ckeditor4 don't override sandbox attribute in iframes by default

mdlama committed:
Distinguish between local params and query params in Solarium 6 (#1812)

This change should ensure that all facets, whether selected for filtering or not, are displayed. Fixes: https://qubeshub.org/support/ticket/5088

denphi committed:
Do not return "OK" message from successful index manipulation (#1817)

mdlama committed:
additional null to '' fixes

denphi committed:
Update display.php

Undefined $contributor->authorid on tools preview

nkissebe committed:
alter a previous committed fix to use slashes rather than backslashes

mdlama committed:
Unnecessary quote in where clause

nkissebe committed:
misc null to '' fixes

nkissebe committed:
check if array before counting

mdlama committed:
Fix file upload for new version of CKEditor

denphi committed:
Update children.php

Typo

mdlama committed:
Update supergroup GitLab to new standards

nkissebe committed:
php 8 null to '' fixes

nkissebe committed:
check if array before foreach-ing it

nkissebe committed:
default get to '' instead of null for php 8

nkissebe committed:
more minor php 8 fixes

nkissebe committed:
fix misspelling of variable name

nkissebe committed:
fix missing gidNumber when doing ldap sync

nkissebe committed:
be sure to pass value as float

nkissebe committed:
more null to '' conditions

nkissebe committed:
fix type and null vs '' issues for php 8

nkissebe committed:
yup another null to '' check

nkissebe committed:
use older symfony/event-dispatcher to resolve deprecations when using mailer

denphi committed:
Fix tool alias validation

denphi committed:
changing to support error  returns

nkissebe committed:
fix group filter issues in support ticket browse list

denphi committed:
Fixed sorting issue

When sorting by username, the join alias should be different

denphi committed:
refactoring get_rank

Avoid double query and sorting index

denphi committed:
Update usage.php

using Joins to force index use

denphi committed:
add controlled external redirection link support

denphi committed:
Update assetgroups.php

missing $ on variable

denphi committed:
Update solr.php

escape solr reserved words

kuang5 committed:
[PURR][#2305]Add a switch to control the display of search category

kuang5 committed:
[PURR][#2305]Add a switch to control the display of search category

denphi committed:
support () on url

support () on url

nkissebe committed:
add null checks in com_citations

nkissebe committed:
check if array before counting

nkissebe committed:
add pear/archive_tar to composer

kuang5 committed:
[PR #1836] Add link to ftp guide for downloading large data

denphi committed:
[PR #1834] Only display published events

nkissebe committed:
only call fastcgi_finish_request if the function exists in the sapi

denphi committed:
[PR #1826] Dont check groups for anonymous users

dependabot[bot] committed:
[PR #1839] Bump phpoffice/phpspreadsheet from 1.29.10 to 1.30.0 in /core

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nkissebe committed:
avoid null to setBcc

nkissebe committed:
Avoid null to strip_tags

denphi committed:
[PR #1827] Disable multiple tickets

denphi committed:
[PR #1840] Fix typo on the commit

nkissebe committed:
missed variable initialization in earlier PR

nkissebe committed:
com_groups admin null to '' check

nkissebe committed:
fallback to system temp directory for htmlpurifier cache

denphi committed:
[PR #1816] fix language string format parameter

nkissebe committed:
loosen check for json data when checking for ini data, leading { without trailing }

nkissebe commented:
Can you re-do this PR against 2.4-main?

pascal-meunier committed:
Fix typo for attr use

shibboleth.php referenced "attr" with a missing $ sign, needs to be $attr

pascal-meunier committed:
Updated shibboleth.php to use newer attributes

eppn is an obsolete field. The code tried to use another obsolete field as replacement. I added more recent attributes as per eduPerson Object Class Specification (202208) v4.4.0
https://wiki.refeds.org/display/STAN/eduPerson+%28202208%29+v4.4.0

pascal-meunier committed:
Attempt to obtain newer attributes in shibboleth.php

Attempt to obtain the newer 'eduPersonPrincipalName' 'eduPersonUniqueId' attributes from a Shibboleth/InCommon login.

nkissebe committed:
use empty string default instead of null

nkissebe committed:
check if directory exists first

nkissebe committed:
null value to '' check

nkissebe committed:
null to '' check in com_dataviewer

nkissebe committed:
default getter to '' instead of null

nkissebe committed:
default getter to '' instead of null

nkissebe committed:
[PURR #2364] manually detect .git repos if not otherwise indicated

nkissebe committed:
fix %2 vs %s in language string format

nkissebe committed:
cron job needs to return true when done or it gets stuck

denphi committed:
Update en-GB.com_answers.ini

denphi committed:
Update questions.php

denphi committed:
Format
